src/pages/dataElements/List.tsx

```diff
@@ -1,64 +1,3 @@
-import { useDataQuery } from '@dhis2/app-runtime'
-import { SharingDialog } from '@dhis2/ui'
-import React, { useEffect } from 'react'
-import { SectionListWrapper } from '../../components'
-import { useModelListView } from '../../components/sectionList/listView'
-import {
-    useSchemaFromHandle,
-    useParamsForDataQuery,
-    DEFAULT_FIELD_FILTERS,
-    DefaultFields,
-} from '../../lib/'
-import { getFieldFilter } from '../../lib/models/path'
-import { Query, WrapQueryResponse } from '../../types'
-import { DataElement, ModelCollectionResponse } from '../../types/models'
+import { DefaultSectionList } from '../DefaultSectionList'
 
-type FilteredDataElement = Pick<DataElement, DefaultFields> &
-    Partial<DataElement>
-
-type DataElements = ModelCollectionResponse<FilteredDataElement, 'dataElements'>
-
-type DataElementsResponse = WrapQueryResponse<DataElements>
-
-const query: Query = {
-    result: {
-        resource: 'dataElements',
-        params: (params) => params,
-    },
-}
-
-export const Component = () => {
-    const { columns, query: listViewQuery } = useModelListView()
-    const initialParams = useParamsForDataQuery()
-    const schema = useSchemaFromHandle()
-    const { refetch, error, data } = useDataQuery<DataElementsResponse>(
-        query,
-        // refetched on mount by effect below
-        { lazy: true }
-    )
-
-    useEffect(() => {
-        // wait to fetch until selected-columns are loaded
-        // so we dont fetch data multiple times
-        if (listViewQuery.isLoading) {
-            return
-        }
-        refetch({
-            ...initialParams,
-            fields: columns
-                .map((column) => getFieldFilter(schema, column.path))
-                .concat(DEFAULT_FIELD_FILTERS),
-        })
-    }, [refetch, initialParams, columns, listViewQuery.isLoading, schema])
-
-    return (
-        <div>
-            <SectionListWrapper
-                error={error}
-                data={data?.result.dataElements}
-                pager={data?.result.pager}
-                refetch={refetch}
-            />
-        </div>
-    )
-}
+export const Component = DefaultSectionList
```

Comment:
I decided to reuse the default list component here. This was because `useDataQuery` would cause issues, and get the wrong data in the list. This happens because the `query` is dynamic, so the querykey resolves to `{ result: { resource: 'dataElements' } }`, which causes `react-query` to resolve the data to the same that is in the `dataElement`-filter, and not the data you want for the list. Essentially leaking data from another source.
Dynamic queries ( params: (params => params)) is very dangerous for this reason, we should not use them.